processing/src/main/java/io/druid/query/topn/AlphaNumericTopNMetricSpec.java

```diff
@@ -38,7 +38,11 @@ public int compare(String str1, String str2)
     {
       int[] pos = {0, 0};
 
-      if (str1.length() == 0) {
+      if (str1 == null) {
+        return str2 == null ? 0 : -1;
```

Comment:
either ways but they should both have same behavior. I'm less inclined to change Lexicographic class because the behavior it gives has been working fine matches up with intuition.